app/metal-controller-manager/internal/server/server.go

```diff
@@ -43,48 +43,86 @@ func (s *server) CreateServer(ctx context.Context, in *api.CreateServerRequest)
 		return nil, err
 	}
 
-	obj := &metal1alpha1.Server{
-		TypeMeta: v1.TypeMeta{
-			Kind:       "Server",
-			APIVersion: metal1alpha1.GroupVersion.Version,
-		},
-		ObjectMeta: v1.ObjectMeta{
-			Name: in.GetSystemInformation().GetUuid(),
-		},
+	obj := &metal1alpha1.Server{}
+
+	if err = c.Get(ctx, types.NamespacedName{Name: in.GetSystemInformation().GetUuid()}, obj); err != nil {
+		if apierrors.IsNotFound(err) {
+			obj := &metal1alpha1.Server{
+				TypeMeta: v1.TypeMeta{
+					Kind:       "Server",
+					APIVersion: metal1alpha1.GroupVersion.Version,
+				},
+				ObjectMeta: v1.ObjectMeta{
+					Name: in.GetSystemInformation().GetUuid(),
+				},
+				Spec: metal1alpha1.ServerSpec{
+					SystemInformation: &metal1alpha1.SystemInformation{
+						Manufacturer: in.GetSystemInformation().GetManufacturer(),
+						ProductName:  in.GetSystemInformation().GetProductName(),
+						Version:      in.GetSystemInformation().GetVersion(),
+						SerialNumber: in.GetSystemInformation().GetSerialNumber(),
+						SKUNumber:    in.GetSystemInformation().GetSkuNumber(),
+						Family:       in.GetSystemInformation().GetFamily(),
+					},
+					CPU: &metal1alpha1.CPUInformation{
+						Manufacturer: in.GetCpu().GetManufacturer(),
+						Version:      in.GetCpu().GetVersion(),
+					},
+				},
+				Status: metal1alpha1.ServerStatus{
+					IsClean: true,
```

Comment:
might need some `//TODO: ` here to change that once we do `Accepted:`

Reply:
Some linter doesn't allow this, I will make a ticket to track.